apps/examples/wifi_manager_sample/wm_test.c

```diff
@@ -262,29 +295,54 @@ void print_wifi_softap_profile(wifi_manager_softap_config_s *config, char *title
 	printf("------------------------------------\n");
 	printf("SSID: %s\n", config->ssid);
 	printf("channel: %d\n", config->channel);
-	/*printf("PASSWORD: %s\n", config->passphrase);*/
 	printf("====================================\n");
 }
 
-
 wifi_manager_ap_auth_type_e get_auth_type(const char *method)
 {
+	char data[20];
+	strcpy(data, method);
+
+	char *result[3];
+	char *next_ptr;
+	result[0] = strtok_r(data, "_", &next_ptr);
+	result[1] = strtok_r(NULL, "_", &next_ptr);
+	result[2] = strtok_r(NULL, "_", &next_ptr);
+
 	int i = 0;
 	int list_size = sizeof(wifi_test_auth_method)/sizeof(wifi_test_auth_method[0]);
 	for (; i < list_size; i++) {
 		if (strcmp(method, wifi_test_auth_method[i]) == 0) {
-			return auth_type_table[i];
+			if (strcmp(result[2], "ent") == 0) {
+				return auth_type_table[i + 2];
+			} else {
+				return auth_type_table[i];
+			}
+		} else if (strcmp(result[0], wifi_test_auth_method[i]) == 0) {
+			if (strcmp(result[2], "ent") == 0) {
+				return auth_type_table[i + 2];
+			} else {
+				return auth_type_table[i];
+			}
 		}
 	}
 	return WIFI_MANAGER_AUTH_UNKNOWN;
 }
 
 wifi_manager_ap_crypto_type_e get_crypto_type(const char *method)
 {
+	char data[20];
+	strcpy(data, method);
+
+	char *result[2];
```

Comment:
actually result is not necessary.

Reply:
Thanks for the comment. That's right, by calling strtok_r(NULL, "_", &next_ptr); again. However thought it does not good for readability and result[1] might be used in the future, when more types of auth/crypto algorithms are added.